pkmodel/solution.py

```diff
@@ -12,6 +18,182 @@ class Solution:
+            t_span=[dose_df.iloc[row_index][0], dose_df.iloc[row_index][1]],
+            y0 = y0, t_eval = np.linspace (dose_df.iloc[row_index][0], dose_df.iloc[row_index][1], 1000)
+        )
+        elif self.model_instance.comp == 2 and self.model_instance.absorb == 1: 
+            sol = scipy.integrate.solve_ivp(
+            fun=lambda t, y: self.rhs_21(row_index,t, y, *arguments),
+            t_span=[dose_df.iloc[row_index][0], dose_df.iloc[row_index][1]],
+            y0 = y0, t_eval = np.linspace (dose_df.iloc[row_index][0], dose_df.iloc[row_index][1], 1000)
+        )
+        else :
+            print("Invalid parameter values")
+
+        return sol
+
+
+    
+    def rhs_10(self, row_index, t, y, absorb, comp, V_c, CL, Q_p1, V_p1, Q_p2, V_p2, k_a):
+        q_c, q_p1 = y
+        transition1 = Q_p1 * (q_c / V_c) - (q_p1 / V_p1)
+        dqc_dt = self.dose(t,row_index) - (q_c / V_c) * CL - transition1 #1 needs replacing with dose
+        dqp1_dt = transition1
+        return [dqc_dt, dqp1_dt]
+
+    def rhs_00(self, row_index, t, y, absorb, comp, V_c, CL, Q_p1, V_p1, Q_p2, V_p2, k_a):
+                q_c = y
+                dqc_dt = self.dose(t,row_index) - (q_c / V_c) * CL
+                return [dqc_dt]
+
+    def rhs_01(self, row_index, t, y, absorb, comp, V_c, CL, Q_p1, V_p1, Q_p2, V_p2, k_a):
+            q_0, q_c = y
+            dq0_dt = self.dose(t,row_index) - k_a * q_0
+            dqc_dt = k_a * q_0 - (q_c / V_c) * CL
+            return [dq0_dt , dqc_dt]
+        
+            
+    def rhs_11(self, row_index, t, y, absorb, comp, V_c, CL, Q_p1, V_p1, Q_p2, V_p2, k_a):
+            q_0, q_c, q_p1 = y
+            transition1 = Q_p1 * (q_c / V_c) - (q_p1 / V_p1)
+            dq0_dt = self.dose(t,row_index) - (k_a * q_0)
+            dqc_dt = (k_a * q_0) - (q_c / V_c) * CL - transition1
+            dqp1_dt = transition1
+            return [dq0_dt , dqc_dt, dqp1_dt]
+            
+
+    def rhs_20(self, row_index, t, y, absorb, comp, V_c, CL, Q_p1, V_p1, Q_p2, V_p2, k_a):
+                q_c, q_p1, q_p2= y
+                transition1 = Q_p1 * (q_c / V_c) - (q_p1 / V_p1)
+                transition2 = Q_p2 * (q_c / V_c) - (q_p2 / V_p2)
+                dqc_dt = self.dose(t,row_index) - (q_c / V_c) * CL - transition1 - transition2
+                dqp1_dt = transition1
+                dqp2_dt = transition2 
+                return [dqc_dt, dqp1_dt, dqp2_dt]
+
+    def rhs_21 (self,row_index, t, y, absorb, comp, V_c, CL, Q_p1, V_p1, Q_p2, V_p2, k_a):
+                q_0, q_c, q_p1, q_p2 = y
+                transition1 = Q_p1 * (q_c / V_c) - (q_p1 / V_p1)
+                transition2 = Q_p2 * (q_c / V_c) - (q_p2 / V_p2)
+                dq0_dt = self.dose(t,row_index) - k_a * q_0
+                dqc_dt = k_a * q_0 - (q_c / V_c) * CL - transition1 - transition2
+                dqp1_dt = transition1
+                dqp2_dt = transition2
+                return [dq0_dt , dqc_dt, dqp1_dt, dqp2_dt]
+
```

Comment:
Would be nice if we could combine all these cases because we have a lot of duplicates!